frontend/components/common/ProductPrice.tsx

```diff
@@ -0,0 +1,79 @@
+import { Box, Text, ThemeTypings, useTheme } from '@chakra-ui/react';
+import { IconBadge } from '@components/ui';
+import { faRectanglePro } from '@fortawesome/pro-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+export type ProductPriceProps = {
+   formattedPrice: string;
+   formattedCompareAtPrice?: string | null;
+   isDiscounted: boolean;
+   discountLabel?: string;
+   showDiscountLabel?: boolean;
+   showProBadge?: boolean;
+   size?: 'large' | 'medium' | 'small';
+   colorScheme?: ThemeTypings['colorSchemes'];
+   direction?: 'row' | 'column' | 'column-reverse';
```

Comment:
Minor: I'm not sure if you're intentionally limiting the `flexDir` types here, but this pattern also works if you weren't.

```suggestion
   direction?: BoxProps['flexDir'];
```